### ISSUE TITLE
prov/gni Expose internal file descriptor from completion queue

### ISSUE THREAD

**Issue by soumagne** [ofiwg/libfabric#4205]
The following operations are currently not supported by the GNI provider:

    cq_attr.wait_obj = FI_WAIT_FD; /* Wait on fd */
    ....
    rc = fi_control(&ctx->noc_cq->fid, FI_GETWAIT, &fd);

In the context of mercury, being able to retrieve the internal GNI file descriptor that gets signaled when something is pushed to the completion queue would be useful in order to add it to a higher level epoll wait set. This would allow us to make progress on different queues or even providers etc from the same thread without busy waiting and wake up when something gets progressed.

**Comment by soumagne** [comment]
For reference, one of the ideas was to add a GNI provider ops_open method on a libfabric CQ to return
the underlying FD’s used by that CQ

**Comment by tenbrugg** [comment]
Cray is interested in this feature as well.

**Comment by shefty** [comment]
Enhancement request with no progress in 2 years.  Assuming low priority and will not address.
